lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -65,10 +65,16 @@ static Value toUnPackLayoutImpl(OpBuilder &builder, Location loc, Value input,
                                           tiles, outerDimsPerm);
 }
 
-static Value handleLayoutNC_NCnc(OpBuilder &builder, Location loc, Value input,
-                                 Value output, ArrayRef<OpFoldResult> tiles) {
+static Value handleLayoutNC_NCnc(OpBuilder &builder, linalg::LinalgOp matmulOp,
+                                 Value input, Value output,
+                                 ArrayRef<OpFoldResult> tiles) {
+  assert(isa<linalg::MatmulOp>(matmulOp) ||
+         isa<linalg::BatchMatmulOp>(matmulOp));
   assert(tiles.size() == 2 && "expect two tile sizes for NC_NCnc");
+  Location loc = matmulOp.getLoc();
   SmallVector<int64_t> innerDimPos = {0, 1};
+  if (isa<linalg::BatchMatmulOp>(matmulOp))
+    innerDimPos = {1, 2};
```

Comment:
The `output` check looks redundant now. Probably true on all other similar methods too.

Reply:
output is used to understand if we need to pack or unpack. The check looks still valid to me.

Reply:
Ignore me, I apparently couldn't read on Friday.